### PR TITLE
dynamically resize the map on no results

### PR DESCRIPTION
**Pull request by oshi97**
This commit dynamically resizes the map when the no results
state appears/disappears. This is because the no results info
that pops up would otherwise cover up part of the map. This
would be a problem primarily if somebody had displayAllResults
set to true for the map, in which case a majority of the pins
would be blocked by alternative verticals suggestions.

J=SLAP-1237
TEST=manual

on no results, the map dynamically resizes
refreshing the page while on no results goes to the list
view, and the map is not visible
desktop looks correct for no results and regular results
"search this area" button is visible in no results

### CODE REVIEW

File: static/js/collapsible-filters/interactions.js [hunk 1]
```diff
@@ -214,7 +214,8 @@ export default class Interactions {
     if (!this._disableScrollToTopOnToggle) {
       this.scrollToTop();
     }
-    ANSWERS.components.getActiveComponent('FilterLink').setState({
+    const filterLink = ANSWERS.components.getActiveComponent('FilterLink');
```
Comment (creotutar) [inline static/js/collapsible-filters/interactions.js:217]: Is this change related?

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 2]
```diff
@@ -478,14 +494,42 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
...
+
   setState(data) {
     if (data.searchState === 'search-loading') {
       return;
     }
 
     this._data = data;
 
-    if (data.resultsContext === 'no-results') {
+    if (data.resultsContext === ResultsContext.NO_RESULTS) {
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:532]: I love the enum

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 3]
```diff
@@ -127,6 +133,12 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
      * @type {Object}
      */
     this.alternativeVerticalsConfig = config.alternativeVerticalsConfig;
+
+    /**
+     * Whether the page is displaying the map on mobile.
+     * @type {boolean}
+     */
+    this.isInMobileMapView = false;
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:141]: This is going to have a conflict with #703 
We may want to update to use the MapStates enum

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 4]
```diff
@@ -24,6 +25,11 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
      */
     this._mapContainerSelector = '#js-answersMap';
 
+    /**
+     * The container in the DOM for the no results block.
+     */
+    this._noResultsContainer = document.querySelector('.js-answersNoResults');
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:31]: Can we query `this._container` instead of `document` to keep it scoped to this component?

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 5]
```diff
@@ -24,6 +25,11 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
      */
     this._mapContainerSelector = '#js-answersMap';
 
+    /**
+     * The container in the DOM for the no results block.
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:29]: nit: type jsdoc please

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 6]
```diff
@@ -478,14 +494,42 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
...
+    const noResults = this._noResultsContainer;
+    const map = document.querySelector(this._mapContainerSelector);
+    if (!noResults || !map) {
+      return;
+    }
+    if (resultsContext === ResultsContext.NO_RESULTS) {
+      const noResultsHeight = noResults.getBoundingClientRect().height;
+      map.style.top =
+        `calc(var(--yxt-maps-mobile-results-header-height) + ${noResultsHeight}px)`;
+      const headerFooterHeight =
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:516]: I think it's a bit confusing because I'd expect it to be an addition but because it's a string interpolation it has to be a subtraction. Was this moved out because the line was too long? I am of the camp that long strings are fine, what do you think?

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 7]
```diff
@@ -478,14 +494,42 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
...
+      return;
+    }
+    const noResults = this._noResultsContainer;
+    const map = document.querySelector(this._mapContainerSelector);
+    if (!noResults || !map) {
+      return;
+    }
+    if (resultsContext === ResultsContext.NO_RESULTS) {
+      const noResultsHeight = noResults.getBoundingClientRect().height;
+      map.style.top =
```
Comment (creotutar) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:514]: I'm a bit worried about having CSS directly applied from the JS. It makes it hard to pinpoint where styling is coming from and haves us worried about two different places for a page's styling when making changes. However I don't see another way because

(1) flex-grow cannot be used as the `mapWrapper` is not sibling to the results header and results footer
(2) the map should be fixed to contain the height on mobile phones

No change request, just voicing the concern

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 3 above)
Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:141]: oh ok I'll wait for 703 to be merged in

Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:141]: then rebase

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 4 above)
Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:31]: yes love that!

File: static/js/collapsible-filters/interactions.js (hunk 1 above)
Comment (oshi97) [reply static/js/collapsible-filters/interactions.js:217]: I was occasionally running into issues with this component not existing (not sure why it would ever not exist, could be related to the map code running before collapsible filters is ready)

Comment (oshi97) [reply static/js/collapsible-filters/interactions.js:217]: I'll make it a separate pr

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 6 above)
Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:516]: hmm, I can split the string up into multiple lines. I'm fine with long strings but not long lines

Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:516]: maybe I would have a different opinion if my monitor was bigger

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 3 above)
Comment (cea2aj) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:142]: 703 has been merged, so you should be good to rebase

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js [hunk 8]
```diff
@@ -340,6 +353,11 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
     this.core.storage.set(StorageKeys.LOCATOR_SELECTED_RESULT, null);
   }
 
+  isMobile() {
```
Comment (cea2aj) [inline static/js/theme-map/VerticalFullPageMapOrchestrator.js:356]: 703 has this exact code, haha

Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:356]: 🧠

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 7 above)
Comment (cea2aj) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:515]: Would it be possible to toggle a css class as usual, and include some new variable, say `--yxt-no-results-height`, which we update from the javascript when we toggle the no results css class?

Comment (cea2aj) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:515]: This is what i'm thinking: https://css-tricks.com/updating-a-css-variable-with-javascript/

Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:515]: oh wow I like that idea a lot, I was trying out a different way of setting the height from what I have in the PR and was running into occasional desync issues (with the update method running before some other random part of the page was done) but I think setting a variable might fix it

File: static/js/theme-map/VerticalFullPageMapOrchestrator.js (hunk 3 above)
Comment (oshi97) [reply static/js/theme-map/VerticalFullPageMapOrchestrator.js:142]: just realized 703 is an area code

**Comment by oshi97** [comment]
talked with @rosiegrant and @creotutar and we decided to not resize the map for no results! future improvement might be to add a close button to the no results pop up (like google maps)
